ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/process/search_tree/Tree.java

```diff
@@ -169,4 +191,48 @@ static RaoResult buildOutput(Leaf rootLeaf, Leaf optimalLeaf) {
         raoResult.setPostOptimVariantId(optimalLeaf.getRaoResult().getPostOptimVariantId());
         return raoResult;
     }
+
+    private static void logOptimalLeaf(Leaf leaf, Crac crac) {
+        LOGGER.info(format("Optimal leaf - %s", generateLeafResults(leaf, crac)));
```

Comment:
I don't know if logging is time consuming, but it might be quite a big log line on "real-size" data with 100 remedial actions. Maybe put a INFO version with the number of RA activated, and a DEBUG version with the details for all RA.

Reply:
so when you're in debug mode you will get the "minimal" line AND the "complete" line following each other ?...

Reply:
I separated NetworkAction and RangeAction information. RangeAction information will only appear at debug level but NetworkAction information is still at info level. But we have to keep in mind that the list of the combined network actions  size is the tree depth and I think for now it won't be higher than a few (like 5). I think we can consider the issue later if thus list becomes very long

Reply:
Yes, I was more thinking of the RangeAction